artifacts/scripts/util.sh

```diff
@@ -990,6 +904,26 @@ checkout-deps-to-kube-commit() {
         pushd ../${dep} >/dev/null
             echo "Checking out k8s.io/${dep} to ${dep_commit}"
             git checkout -q "${dep_commit}"
+
+            echo "Downloading go mod dependencies..."
+            go mod download
+
+            local pseudo_version=$(gomod-pseudo-version)
+            local cache_dir="${GOPATH}/pkg/mod/cache/download/${base_package}/${dep}/@v"
```

Comment:
I'd be hesitant to modify the modcache directory with our own "fake published" module versions. Would it be possible to do something like this instead:
1. run `go mod download` on k/k to fetch all non-staging dependencies into the module cache
2. copy content of `${GOPATH}/pkg/mod/cache/download` to `/some/goproxy/dir`
3. fake publish modules to `/some/goproxy/dir/${base_package}/${dep}/@v`
4. run module operations with `GOPROXY=file:///some/goproxy/dir`